crates/bytecode/src/compiler.rs

```diff
@@ -263,7 +263,7 @@ impl Compiler {
                 result
             }
             Node::Nested(nested) => self.compile_node(*nested, ctx)?,
-            Node::Id(index) => self.compile_load_id(*index, ctx)?,
+            Node::Id(index, ..) => self.compile_load_id(*index, ctx)?,
```

Comment:
It's interesting that the `..` syntax here works when there aren't any additional fields in `Node::Id` (and later `ChainNode::Id`) - I think for the parsing to be complete, the IDs need to be connected to their type hints. `Node::Id` should have an `Option<AstIndex>` that refers to the type hint. In the current state the type hint gets added in to the AST in a disconnected state. I'll add a comment on the test to explain further.